velox/vector/tests/utils/VectorMaker.h

```diff
@@ -605,6 +605,72 @@ class VectorMaker {
     return arrayVectorNullable<T>(arrays, arrayType);
   }
 
+  /// Creates an ArrayVector from a list of JSON arrays of arrays.
+  ///
+  /// JSON arrays of arrays can represent a null array, an empty array or array
+  /// with null elements.
+  ///
+  /// Examples:
+  /// [[1, 2], [2, 3, 4], [null, 7]]
+  /// [[1, 3, 7, 9], []]
+  /// [] - empty array of arrays
+  /// null - null array of arrays
+  /// [null]
+  /// [[]]
+  ///
+  /// @tparam T Type of array elements.
+  /// @param jsonArrays A list of JSON arrays. JSON array cannot be an empty
+  /// string.
+  template <typename T>
+  ArrayVectorPtr nestedArrayVectorFromJson(
+      const std::vector<std::string>& jsonArrays,
+      const TypePtr& arrayType = ARRAY(CppToType<T>::create())) {
+    std::vector<std::optional<std::vector<std::optional<T>>>> baseVector;
+    std::vector<vector_size_t> offsets;
+    std::vector<vector_size_t> nulls;
+    int offset = 0;
+    int nullIdx = 0;
+    for (const auto& jsonArray : jsonArrays) {
+      VELOX_CHECK(!jsonArray.empty());
+      const folly::dynamic arraysObject = folly::parseJson(jsonArray);
+      offsets.push_back(offset);
+      if (arraysObject.isNull()) {
+        // Null array.
+        nulls.push_back(nullIdx);
+        continue;
+      }
+      if (arraysObject.size() == 0) {
+        // Empty nested array
+        std::vector<std::optional<T>> empty;
+        baseVector.push_back(empty);
+        offset++;
+        continue;
+      }
+
+      for (const auto& nestedArray : arraysObject) {
+        if (nestedArray.isNull()) {
```

Comment:
It might be cleaner to write this as an if..else and avoid the continue

```
if (nestedArray,isNull()) {
   // Null nested array.
   baseVector.push_back(std::nullopt);
} else {
   std::vector<std::optional<T>> elements;
    for (const auto& element : nestedArray) {
      if (element.isNull()) {
        // Null element.
        elements.push_back(std::nullopt);
      } else {
        elements.push_back(detail::jsonValue<T>(element));
      }
    }
    baseVector.push_back(elements);
}
```